lesson18/script.js

```diff
@@ -0,0 +1,186 @@
+    const fragment = document.createDocumentFragment();
+    for(const image of sliderData){
+        const sliderItem = createAttributedElements({tag:"div",attrObj:{class:"slider__item"}});
+        const slideImage = createAttributedElements({tag:"img",attrObj:{class:"slide-image",src:image.img,alt:image.alt}});
+        
+        fragment.appendChild(sliderItem).appendChild(slideImage);
+    }
+
+    slider.appendChild(fragment);
+}
+
+const getPageNumber = (sliderData) => {
+    const pageNumber    = createAttributedElements({tag:"p",attrObj:{class:"page-number"},str:`/${sliderData.length}`});
+    const currentNumber = createAttributedElements({tag:"span",attrObj:{id:"js-currentNumber",class:"current-number"},str:"1"});
+
+    slider.appendChild(pageNumber).insertAdjacentElement("afterbegin",currentNumber);
+}
+
+const renderButtons = (sliderData) => {
+    const prevBtn = createAttributedElements({tag:"button",attrObj:{id:"js-prevBtn",class:"js-btn prev-btn slider-btn",disabled:true}});
+    const nextBtn = createAttributedElements({tag:"button",attrObj:{id:"js-nextBtn",class:"js-btn next-btn slider-btn"}});
+
+    slider.appendChild(prevBtn).insertAdjacentElement("afterend",nextBtn);
+    addClickEventForButton(sliderData);
+}
+
+const renderDotPagination = (sliderData) => {
+    const paginationList = createAttributedElements({tag:"ul",attrObj:{id:"js-dot-pagination",class:"dot-pagination"}});
+
+    const fragment = document.createDocumentFragment();
+    const length   = sliderData.length;
+    for(let i = 0; i < length; i++){
+        const paginationItem = createAttributedElements({tag:"li",attrObj:{class:"dot-pagination__item"}});
+        fragment.appendChild(paginationItem);
+    }
+
+    slider.appendChild(paginationList).appendChild(fragment);
+    addClickEventForDotPagination(sliderData);
+}
+
+let currentIndex = 0;
+const addClickEventForButton = (sliderData) => {
+    const sliderButtons = document.getElementsByClassName("js-btn");
+
+    for (const button of sliderButtons ){
+        button.addEventListener("click",function(){
+            this.id === "js-nextBtn" ? currentIndex++ : currentIndex--;
+
+            switchSlider(sliderData,currentIndex);
+            resetAutoPlay(sliderData);
+        });
+    }
+}
+
+const addClickEventForDotPagination = (sliderData) => {
+    const dotPaginetions   = document.getElementsByClassName("dot-pagination__item");
+    const arrayPaginations = [...dotPaginetions];
+    
+    for(const dotPagination of dotPaginetions ){
+        dotPagination.addEventListener("click",function(){
+            currentIndex = arrayPaginations.indexOf(this);
+
+            switchSlider(sliderData,currentIndex);
+            resetAutoPlay(sliderData);
+        });
+    }
+}
+
+const isActiveFirstItem = () => {
+    document.getElementsByClassName("slider__item")[0].classList.add("is-active");
+    document.getElementsByClassName("dot-pagination__item")[0].classList.add("is-active");
```

Comment:
IMO
Also, I think there will be several opportunities to get `slider__item` and `dot-pagination__item` on js, 
but since they are both CSS-applied, I thought it would be better to separate their roles with a js-prefix.
Because it will be affected when the CSS class name changes.

----

nits
this may be simpler
```suggestion
    document.querySelector(".slider__item").classList.add("is-active");
    document.querySelector(".dot-pagination__item").classList.add("is-active");
```

Reply:
Certainly this is very simple and nice!
I fixed.
https://github.com/nakagawamai/js-lesson/pull/20/commits/ef2eec12818f75ad8ecd411aea860b520740eeda